### PR TITLE
ENH+RF: Share 'pip {list,show}' code between venv and conda tracers

### PR DESCRIPTION
**Pull request by kyleam**
Adjust the conda tracer to reuse the batched `pip show` from the venv
tracer (#162).

Note that this is not ready to be merged because it depends on #162.
Currently this PR is against a temporary branch rather than master so
that the diff doesn't include the commits from #162.

### CODE REVIEW

**Comment by kyleam** [comment]
Underneath the batching is done using @rbuccigrossi's
`execute_command_batch` from #153, which works nicely.  Thanks :)

**Comment by kyleam** [comment]
The two commits just pushed broaden the scope of this issue a bit in
order to fix #163.

File: niceman/distributions/conda.py [hunk 1]
```diff
@@ -157,57 +157,18 @@ def _get_conda_package_details(self, conda_path):
...
-                            "export from path %s: %s", conda_path,
-                            exc_str(exc))
-                continue
-
+    def _get_conda_pip_package_details(self, conda_path):
+        pip = conda_path + "/bin/pip"
+        try:
+            pkgs = list(pip_packages(self._session, pip))
+        except Exception as exc:
+            lgr.warning("Could determine pip packages for %s: %s",
```
Comment (yarikoptic) [inline niceman/distributions/conda.py]: I guess "Could **not** determine..."?

File: niceman/distributions/venv.py [hunk 2]
```diff
@@ -72,26 +71,14 @@ def _create_package(self, **package_fields):
...
-            for path in info["Files"]:
-                full_path = os.path.normpath(
-                    os.path.join(info["Location"], path))
-                file_to_pkg[full_path] = pkg
-        return packages, file_to_pkg
+        pip = venv_path + "/bin/pip"
+        try:
+            pkgs = list(pip_packages(self._session, pip))
+        except Exception as exc:
+            lgr.warning("Could determine pip packages for %s: %s",
```
Comment (yarikoptic) [inline niceman/distributions/venv.py]: the same "Could **not**"

File: niceman/distributions/piputils.py [hunk 3]
```diff
@@ -38,6 +42,50 @@ def parse_pip_show(out):
...
 
 
+def _pip_batched_show(session, which_pip, pkgs):
+    cmd = [which_pip, "show", "-f"]
+    batch = execute_command_batch(session, cmd, pkgs)
+    entries = (stacked.split("---") for stacked, _, _ in batch)
+
+    for pkg, entry in zip(pkgs, itertools.chain(*entries)):
+        info = parse_pip_show(entry)
+        yield pkg, info
```
Comment (yarikoptic) [inline niceman/distributions/piputils.py]: I would have appreciated a unittest for this function which mocks away `execute_command_batch` and demonstrate that it does work e.g. when output for multiple pkgs in a batch is output (on some dummified but still legit).  That could allow others comprehend on what is going on here (how pip show -f output should look like etc), if things start to break (e.g. `pip show -f` format changes or smth odd in it, like someone using `---` in the Summary or a url for some reason ;-))

File: niceman/distributions/conda.py (hunk 1 above)
Comment (kyleam) [reply niceman/distributions/conda.py]: guh, thanks

**Comment by yarikoptic** [comment]
the `---` will be dealt with later, merging

**Comment by kyleam** [comment]
> the --- will be dealt with later, merging

Addressed with #180.

**Pull request merged**